aws/resource_aws_athena_named_query_test.go

```diff
@@ -0,0 +1,85 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/athena"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSAthenaNamedQuery_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAthenaNamedQueryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAthenaNamedQueryConfig(acctest.RandInt(), acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAthenaNamedQueryExists("aws_athena_named_query.foo"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAthenaNamedQueryDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).athenaconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_athena_named_query" {
+			continue
+		}
+
+		input := &athena.GetNamedQueryInput{
+			NamedQueryId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetNamedQuery(input)
+		if err != nil {
+			if isAWSErr(err, athena.ErrCodeInvalidRequestException, "does not exist") {
```

Comment:
I'm not sure what's going on here, but the test is failing for me:

```
=== RUN   TestAccAWSAthenaNamedQuery_basic
--- FAIL: TestAccAWSAthenaNamedQuery_basic (104.99s)
	testing.go:563: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Check failed: InvalidRequestException: fb403822-2e67-4344-b530-fc488aa79c49
			status code: 400, request id: aa8710fc-c5ff-11e7-97cf-5fde2fdbf5a4

		State: <no state>
```
Is it passing for you? 🤔